### PR TITLE
Adjust docs development workflow and add developers' README

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** This PR adds a setup target to docs Makefile and modifies POETRY variable to ensure compatibility for windows, as well as a README for developers as requested in https://github.com/scylladb/scylla-operator/pull/1308#issuecomment-1669262385

**Which issue is resolved by this Pull Request:**
Resolves: issues https://github.com/scylladb/scylla-operator/pull/1308#issuecomment-1669262385

### CODE REVIEW

**Comment by rzetelskik** [comment]
cc @dgarcia360, I can't request a review for some reason

**Comment by dgarcia360** [comment]
LGTM

File: docs/README-dev.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Building the docs
+
+This project uses [Sphinx Theme](https://sphinx-theme.scylladb.com/) , but we have modified the base project to adapt it to our workflows.
+
+## Local preview
+
+To build the documentation locally, run `make setup` first.
```
Comment (tnozicka) [inline docs/README-dev.md:7]:
```suggestion
To build the documentation locally, run `make setup` first to install its dependencies.
```

**Review by tnozicka — APPROVED** [review]

**Review by zimnx — APPROVED** [review]

File: docs/README-dev.md (hunk 1 above)
Comment (rzetelskik) [reply docs/README-dev.md]: done

**Review by zimnx — APPROVED** [review]

**Pull request merged**